### DISCUSSION TITLE
请问怎么用Airsim_ros 跑orbslam2或者vins？

### DISCUSSION THREAD

**Discussion by SharineLee** [GimpelZhang/airsim_ros#16] [Q&A]
主要的架构思路和需要修改的文件可以说说么？谢谢

**Comment by GimpelZhang** [comment]
用airsim的这个ros外壳，运行双目视觉算法的话，主要问题会在于双目时间戳差别过大。解决这个问题我当时参考的是：[xuhao1/airsim_ros_pkgs](https://github.com/xuhao1/airsim_ros_pkgs)，我认为主要的修改内容可能在于airsim_ros_wrapper.cpp文件中void AirsimROSWrapper::create_ros_pubs_from_settings_json()这一函数体里的current_image_request_vec这一向量的声明位置：[链接](https://github.com/GimpelZhang/airsim_ros/blob/d56af00556a27465e6a9cc61e324acd36f7ea3ff/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp#L189-L191)从底下的for循环中提取了出来。
时间隔得有点久我记得可能已经不对了。另外这项修改测试通过的版本也停留在了2020年9月4日，后面的master更新我都没有测试过。
